packages/pwa-kit-extension-sdk/src/react/utils/helpers.ts

```diff
@@ -23,3 +23,40 @@ export const applyHOCs = <T extends React.ComponentType<any>>(Component: T, hocs
         return hoistNonReactStatics(WrappedComponent, AccumulatedComponent) as T
     }, Component)
 }
+
+/**
+ * Applies a cache to a method on an instance. The cache is stored in a property on the instance.
+ *
+ * @param instance - The instance on which to apply the cache.
+ * @param methodName - The name of the method to cache.
+ * @param cacheProperty - The name of the property to use for caching.
+ */
+export function cacheMethodResult(instance: any, methodName: string, cacheProperty: string) {
+    const originalMethod = instance[methodName]
+    if (typeof originalMethod !== 'function') return
+
+    instance[methodName] = function (...args: any[]) {
+        if (instance[cacheProperty] !== undefined && instance[cacheProperty] !== null) {
+            return instance[cacheProperty]
+        }
+
+        const result = originalMethod.apply(instance, args)
+
+        if (result instanceof Promise) {
+            result.then((resolved) => {
+                instance[cacheProperty] = resolved
+            })
+        } else {
+            instance[cacheProperty] = result
+        }
```

Comment:
i'm not sure exactly why but I find the unit test fails when checking if the value has been cached when it is missing the following code to handle the sync case. 

`npm run test packages/pwa-kit-extension-sdk/src/react/utils/helpers.test.tsx`

Reply:
@hajinsuha1 yes, thank you.. sorry I forgot to immediately save the result when it's not a promise 👍